### PR TITLE
[8.x] Add new test case of withCount

### PR DESCRIPTION
**Pull request by tu6ge**
`withCount` method is not only compute count of relate rows , It can compute sum or avg of relate fields

### CODE REVIEW

File: tests/Database/DatabaseEloquentBuilderTest.php [hunk 1]
```diff
@@ -845,6 +846,50 @@ public function testWithCountMultipleAndPartialRename()
...
+
+        $this->assertSame('select "eloquent_builder_test_model_parent_stubs".*, (select sum(num1) from "eloquent_builder_test_model_close_related_stubs" where "eloquent_builder_test_model_parent_stubs"."foo_id" = "eloquent_builder_test_model_close_related_stubs"."id") as "foo_count" from "eloquent_builder_test_model_parent_stubs"', $builder->toSql());
+    }
+
+    public function testWithCountSumRename()
+    {
+        $model = new EloquentBuilderTestModelParentStub;
+
+        $builder = $model->withCount(['foo as foo_sum'=>function($query){
+            $query->where(DB::raw('sum(num1)'));
```
Comment (GrahamCampbell) [inline tests/Database/DatabaseEloquentBuilderTest.php:865]: I don't think we need raw to achieve this do we?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Database/DatabaseEloquentBuilderTest.php (hunk 1 above)
Comment (tu6ge) [reply tests/Database/DatabaseEloquentBuilderTest.php:865]: Maybe you're right

Comment (tu6ge) [reply tests/Database/DatabaseEloquentBuilderTest.php]: Raw  needs to be used

**Comment by taylorotwell** [comment]
Tests are failing.

**Comment by tu6ge** [comment]
I am in china, Just wake up to see this pr is closed

**Comment by GrahamCampbell** [comment]
If you submit a new PR with passing tests, it will be considered. This is always the case for PRs closed on that basis.

**Pull request closed without merging**